### PR TITLE
storage,sql: fix naming nits in `mz_source_statistics`

### PR DESCRIPTION
**Pull request by benesch**
Just some naming nits I noticed recently. In the future the SQL council should make the naming review happen smoothly as new relations are added.

----

To comport with the system catalog style guide:

  * Change `rehydration_latency_ms uint8` to `rehydration_latency interval`, since interval types are preferred to integers where the units are indicated in the column name's suffix.

    This rule was undocumented, so also add it to the style guide.

  * Rename `envelope_state_count` to `envelope_state_records`. This one is more subjective, but "records" aligns with names used in other relations, like `mz_dataflow_arrangement_sizes.records` and `mz_records_per_dataflow`.

### Motivation

   * This PR refactors existing code.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [x] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [x] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [x] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - Change the `mz_internal.mz_source_statistics` as follows:
    - Rename the `envelope_state_count` column to `envelope_state_records`.
    - Replace the column named `rehydration_latency_ms` with type `uint8` with a new column named `rehydration_latency` and type `interval`.

### CODE REVIEW

File: src/storage/src/statistics.rs [hunk 1]
```diff
@@ -71,7 +71,8 @@ impl SourceStatisticsMetricsDefinitions {
                 help: "The number of bytes of the source envelope state kept. This will be specific to the envelope in use.",
                 var_labels: ["source_id", "worker_id", "parent_source_id", "shard_id"],
             )),
-            envelope_state_count: registry.register(metric!(
+            envelope_state_records: registry.register(metric!(
+                // The different name for the metric itself is historical.
                 name: "mz_source_envelope_state_count",
```
Comment (antiguru) [inline src/storage/src/statistics.rs:76]: In Compute, I'd prefer to change the metric name, too. It's relatively simple to work-around the name change in Grafana, we just need to be aware where it's used from and duplicate the query with the new name. But up to you!

File: src/storage-client/src/client.proto [hunk 2]
```diff
@@ -82,6 +82,7 @@ message ProtoStorageCommand {
 
 message ProtoStorageResponse {
     message ProtoSourceStatisticsUpdate {
+        reserved 10;
```
[the remark this answers is not in the document]
Comment (benesch) [reply src/storage-client/src/client.proto:85]: Ah, thanks, fixed.

File: src/storage/src/statistics.rs (hunk 1 above)
Comment (benesch) [reply src/storage/src/statistics.rs]: Ack, I've gone ahead and done that. @moulimukherjee / @guswynn: let me know if you'd prefer the original approach of leaving the metric name unchanged.

**Review by morsapaes — APPROVED** [review]

**Review by ParkMyCar — APPROVED** [review]
Looks good to me, thank you for updating the style guide too!

File: src/storage/src/statistics.rs (hunk 1 above)
Comment (guswynn) [reply src/storage/src/statistics.rs]: I prefer it all to match!

Comment (benesch) [reply src/storage/src/statistics.rs]: Great!

**Pull request merged**